src/sentry/silo/util.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+import hmac
+from hashlib import sha256
+from typing import Any, Mapping
+
+from django.conf import settings
+
+INVALID_PROXY_HEADERS = ["Host"]
+
+
+def clean_proxy_headers(headers: Mapping[str, Any] | None) -> Mapping[str, Any]:
+    if not headers:
+        headers = {}
+    modified_headers = {**headers}
+    for invalid_header in INVALID_PROXY_HEADERS:
+        modified_headers.pop(invalid_header, None)
+    return modified_headers
+
+
+def encode_subnet_signature(
+    secret: str,
+    timestamp: str,
+    request_body: str | bytes,
+):
+    """v0: Silo subnet signature encoding"""
+    raw_signature = b"v0|%s|%s" % (timestamp.encode("utf-8"), request_body)
+    signature = hmac.new(secret.encode("utf-8"), raw_signature, sha256).hexdigest()
+    return f"v0={signature}"
+
+
+def verify_subnet_signature(
+    timestamp: str,
+    request_body: bytes,
+    provided_signature: str,
+) -> bool:
+    """v0: Silo subnet signature decoding and verification"""
+    secret = getattr(settings, "SENTRY_SUBNET_SECRET")
+    if not secret:
+        return False
+
+    assembled_signature = encode_subnet_signature(secret, timestamp, request_body)
```

Comment:
I don't think we can reliably use timestamps in our signatures. We will experience clock drift between regions and operations could easily span between seconds which would cause signature failures. Using an HMAC of the message body should be enough. If you wanted to lock it down more than just the shared key, you could include the organizationintegration id, and path that is being proxied.

Reply:
@markstory That makes sense, the signatures are being verified using timestamps that are provided in the headers though, so it's just ensuring consistency in the request as sent, I haven't added a window or something to ensure the control's timestamp is within X minutes just yet.

I guess if we're not doing that check though it might be unnecessary to keep, lmk your thoughts but I will be adding the path and organizationintegration id, that sounds good.